packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,395 @@
+// import { TypeSpecConfig } from "@typespec/compiler";
+import path, { dirname } from "path";
+import vscode, { Uri } from "vscode";
+import { Executable } from "vscode-languageclient/node.js";
+import { StartFileName } from "../const.js";
+import logger from "../log/logger.js";
+import { InstallationAction, npmDependencyType, NpmUtil } from "../npm-utils.js";
+import {
+  getMainTspFile,
+  resolveTypeSpecCli,
+  toError,
+  toOutput,
+  TraverseMainTspFileInWorkspace,
+} from "../typespec-utils.js";
+import { ExecOutput, isFile, spawnExecution } from "../utils.js";
+import { EmitQuickPickItem, TypeSpecProjectPickItem } from "./emit-quick-pick-item.js";
+import { Emitter, EmitterKind, getRegisterEmitters } from "./emitter.js";
+
+export async function doEmit(
+  context: vscode.ExtensionContext,
+  mainTspFile: string,
+  kind: EmitterKind,
+  overallProgress: vscode.Progress<{ message?: string; increment?: number }>,
+) {
+  if (!mainTspFile || !(await isFile(mainTspFile))) {
+    logger.info(
+      "Invalid typespec project. There is no main tsp file in the project. Generating Cancelled.",
+      [],
+      { showOutput: false, showPopup: true, progress: overallProgress },
```

Comment:
the progress option has been removed and will be ignored by logger now. You will need to either explicitly update the msg in progress or remove it if it's not needed which is current behavior...

Reply:
removed.